trpcultivate_germcollection/src/Plugin/TripalImporter/GermplasmRelationshipImporter.php

```diff
@@ -0,0 +1,1189 @@
+            if (in_array($line_text . $val_uniqname, $temp_lines)) {
+              // Inspect if this name has same type, ogranism and uniquename.
+              $duplicate_line = array_search($line_text . $val_uniqname, $temp_lines);
+              throw new \Exception('Duplicate in lines: #' . $duplicate_line . ' and #' . ($i + 1));
+            }
+            else {
+              if (preg_grep("/$line_text.*/", $temp_lines)) {
+                $duplicate[] = $val_name;
+              }
+
+              // Record instance.
+              $temp_lines[$i + 1] = $line_text . $val_uniqname;
+            }
+
+            // Uniquename:
+            // Check if the uniquename already exists in the database.
+            // If it does, throw an exception.
+            if (!$population['relationship_only']) {
+              if ($val_uniqname) {
+                $query = $this->chado_connection->select('1:stock', 's')
+                  ->fields('s', ['stock_id'])
+                  ->condition('s.uniquename', $val_uniqname, '=')
+                  ->execute();
+
+                $result = NULL;
+                if ($stock_id = $query->fetchField()) {
+                  $result = $stock_id;
+                }
+                if ($result) {
+                  // A uniquename is already used in database.
+                  throw new \Exception('Uniquename is already used by another germplasm.');
+                }
+                else {
+                  // Check if the uniquename is duplicated
+                  // in the file. If it is, throw an exception.
+                  if (in_array($val_uniqname, $temp_uname)) {
+                    // Duplicate uniquename in file.
+                    $duplicate_uname = array_search($val_uniqname, $temp_uname);
+                    throw new \Exception('Duplicate Uniquename in lines: #' . $duplicate_uname . ' and ' . ($i + 1));
+                  }
+                  else {
+                    // If uniquename is not duplicated, add it to the temp array
+                    // to check against for future lines in the file.
+                    $temp_uname[$i + 1] = $val_uniqname;
+                  }
+                }
+              }
+
+              // Stock Name+Type+Scientific Name combination must be unique.
+              // Check if the combination already exists in the database.
+              // If it does, throw an exception.
+              $query = $this->chado_connection->select('1:stock', 's')
+                ->fields('s', ['stock_id'])
+                ->condition('s.name', $val_name, '=')
+                ->condition('s.type_id', $type_id, '=')
+                ->condition('s.organism_id', $organism_id, '=')
+                ->execute();
+              $result = NULL;
+              if ($stock_id = $query->fetchField()) {
+                $result = $stock_id;
+              }
+              if ($result) {
+                throw new \Exception('Term already exists in the database.');
```

Comment:
"Term" is too vague here. I would reword this similarly to how you worded the error on line 906:
```suggestion
                throw new \Exception('Germplasm with name: ' . $val_name . ' + type: ' . $val_type . ' + scientific name: ' . $val_sciname . ' already exists in the database, but the toggle was set to create new individuals.');
```